microcosm_postgres/tests/test_company_store.py

```diff
@@ -93,6 +93,22 @@ def test_create_delete_company(self):
             raises(ModelNotFoundError, pattern="Company not found"),
         )
 
+    def test_create_delete_company_complicated_expression(self):
+        """
+        Should not be able to retrieve a company after deleting it.
+
+        """
+        with transaction():
+            company = Company(name="name").create()
+
+        with transaction():
+            self.company_store._delete(Company.name.in_(["name"]), synchronize_session="fetch")
```

Comment:
The `in_` expression is too complicated for the `evaluate` expression to handle it, so you have to use the fetch expression, or explicitly use the `_query` object.